src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ExplosionsListener.java

```diff
@@ -50,18 +53,33 @@ public void onBlockExplode(BlockExplodeEvent e) {
         removeResistantBlocks(e.blockList().iterator());
     }
 
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onEntityBreak(EntityChangeBlockEvent e) {
+        if (e.getEntity().getType() == EntityType.WITHER || e.getEntity().getType() == EntityType.WITHER_SKULL) {
+            removeResistantBlock(e.getBlock());
+        }
+    }
+
+    private void removeResistantBlock(@Nonnull Block block) {
+        final SlimefunItem item = BlockStorage.check(block);
+
+        if (item != null
+            && !(item instanceof WitherProof)
+            && !item.callItemHandler(BlockBreakHandler.class, handler -> handleExplosion(handler, block))
+        ) {
+            BlockStorage.clearBlockInfo(block);
+            block.setType(Material.AIR);
+        }
+    }
+
     private void removeResistantBlocks(@Nonnull Iterator<Block> blocks) {
         while (blocks.hasNext()) {
             Block block = blocks.next();
             SlimefunItem item = BlockStorage.check(block);
 
             if (item != null) {
                 blocks.remove();
-
-                if (!(item instanceof WitherProof) && !item.callItemHandler(BlockBreakHandler.class, handler -> handleExplosion(handler, block))) {
-                    BlockStorage.clearBlockInfo(block);
-                    block.setType(Material.AIR);
-                }
+                removeResistantBlock(block);
```

Comment:
This method already performs the pretty heavy lookup `BlockStorage.check`. So doing it twice FOR EVERY SINGLE block here is a huge performance drawback.